### PR TITLE
feat(docs): Add cdk version, metrics, and last updated

### PR DESCRIPTION
**Pull request by bnchrch**
## What

Adds the following to the metrics component in docs
1. CDK version
2. Last updated ad
3. Sync success rate
4. Usage rate

Does not
1. Add metric icons. (Next PR

### CODE REVIEW

**Comment by bnchrch** [comment]
* **#38799** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38799?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by lmossman — CHANGES_REQUESTED** [review]
Had some questions, suggestions, and nits. The things that aren't rendering properly should at least be fixed first before approving

File: docusaurus/src/remark/docsHeaderDecoration.js [hunk 1]
```diff
@@ -25,6 +27,13 @@ const plugin = () => {
         const originalTitle = node.children[0].value;
         const originalId = node.data.hProperties.id;
 
+        const rawCDKVersion = getFromPaths(registryEntry, "packageInfo_[oss|cloud].cdk_version");
+        const syncSuccessRate = getFromPaths(registryEntry, "generated_[oss|cloud].metrics.cloud.sync_success_rate");
+        const usageRate = getFromPaths(registryEntry, "generated_[oss|cloud].metrics.[all|cloud|oss].usage");
```
[the remark this answers is not in the document]
Comment (bnchrch) [reply docusaurus/src/remark/docsHeaderDecoration.js:32]: > Does the generated_oss metrics object actually have a .cloud.usage path? Or is this just a side-effect of the path unioning logic?

**Some context:**

1. `generated_oss` relates directly to the `generated` field in the OSS Registry
2. `generated_cloud` relates directly to the `generated` field in the Cloud Registry 

This is just a quirk of how we transform the 2 registries into one registry report for docs to use. So that we have 
the union of both catalogs.

Also the reason we have all vs oss vs cloud for metrics is simply because "we had the data in metrics"

We should always be using `all`, `all` should always have data in their respective registries.

**Answer**
generated_oss metrics does have a .cloud.usage path. But its really there as a backup, and `all` should always have data.

> Why are the usage metrics split up into separate paths like this?
Basically because we could and since we could may as well add them to make any future request easier.

Which became immediately useful when Natik asked if we could skip tests for low to none cloud usage.

> Does this mean we are always trying to show usage across all deployments if possible, but if not then only across cloud deployments, and if not that then only across oss deployments?
Yes

[the remark this answers is not in the document]
Comment (lmossman) [reply docusaurus/src/remark/docsHeaderDecoration.js:32]: Got it. If we expect `all` to usually always have data then this seems fine to me.

File: docusaurus/src/components/HeaderDecoration.jsx [hunk 2]
```diff
@@ -91,11 +154,28 @@ export const HeaderDecoration = ({
...
+  cdkVersionUrl,
+  syncSuccessRate,
+  usageRate,
+  lastUpdated,
 }) => {
   const isOss = isOssString.toUpperCase() === "TRUE";
   const isCloud = isCloudString.toUpperCase() === "TRUE";
   const isPypiPublished = isPypiPublishedString.toUpperCase() === "TRUE";
   const isArchived = supportLevel.toUpperCase() === "ARCHIVED";
+  const isLatestCDK = isLatestCDKString.toUpperCase() === "TRUE";
```
[the remark this answers is not in the document]
Comment (bnchrch) [reply docusaurus/src/components/HeaderDecoration.jsx]: Ahh yes. Ok big miss on my part. `?.` shouldve been there.

Just fixed this by adding a couple helper functions with a small explanation as to why we have to use strings instead of straight booleans

File: docusaurus/src/components/HeaderDecoration.jsx [hunk 3]
```diff
@@ -34,7 +39,40 @@ const EnabledIcon = ({ isEnabled }) => {
...
+    .replace(" second", "s")
+    .replace(" days", "d")
+    .replace(" day", "d")
+    .replace(" weeks", "w")
+    .replace(" week", "w")
+    .replace(" months", "m")
+    .replace(" month", "m")
+    .replace(" years", "y")
+    .replace(" year", "y");
+};
```
[the remark this answers is not in the document]
Comment (bnchrch) [reply docusaurus/src/components/HeaderDecoration.jsx]: Good catch. Done

File: docusaurus/src/components/HeaderDecoration.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bnchrch) [reply docusaurus/src/components/HeaderDecoration.jsx]: > Separate question: why are isOss, isCloud, isPypiPublished, and isLatestCDKString string values instead of boolean values? Seems like booleans would make more sense here.

So! This was a weird system implementation detail I had to discover.

The way remark passes these values to the <HeaderDecoration /> is through raw html attributes it adds to an element during generation.

This leads to some odd behaviour where
1. boolean values aren't translate properly
2. undefineds turn into true

What I did to address is 
1. filter out undefined values before appending to the html element
2. created two helper functions for translating between boolean strings

> Looking [here](https://github.com/airbytehq/airbyte/pull/38799/files#diff-c8c2d902770fab687d48ecc1481bb45950ff556cdb56a0cc63f52bc05ba10ac2R42-R44) it seems those string values are coming from the registy, but I think we should convert them to boolean values in that connector_registry.js file so that this component can receive nice boolean values

This is a read detail, as the value is correct in the report. 
image
So it seems like `true` in the registry report becomes `"true"` when we process it.
Likely a bug in how we read the file.

But Im not confident I can remediate that without accidentally introducing another bug in our docs as it seems like we have other code that relies on it being a Boolean String.

File: docusaurus/src/helpers/objects.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  // Check if options are integers
+  // If so, return the original string
+  // This is to maintain compatibility with arrays
+  if (/^\d+$/.test(options)) {
+    return [str];
+  }
+
+  const splitOptions = options.split("|");
+  const prefix = str.substring(0, match.index);
+  const suffix = str.substring(match.index + match[0].length);
```
[the remark this answers is not in the document]
Comment (bnchrch) [reply docusaurus/src/helpers/objects.js:31]: I think youre right. Im not going to implement this now. Since its working and I desperately want to get back to some platform code.

LMK if this NIT, becomes a DO

File: docusaurus/src/components/HeaderDecoration.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmossman) [reply docusaurus/src/components/HeaderDecoration.jsx]: I see. Maybe if we didn't use remark for this we could avoid that problem, but I don't think its worth refactoring it just for that. Having a comment is fine :+1:

File: docusaurus/src/helpers/objects.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lmossman) [reply docusaurus/src/helpers/objects.js:31]: Nope this isn't blocking, just wanted to comment it since I noticed it

**Review by lmossman — APPROVED** [review]
Changes LGTM. Seems like there is a pnpm-lock.yaml conflict - probably just need to rebase on master and re-install `dayjs`, but I don't think that needs another review after being fixed.

EDIT: I pushed up a super small commit to add some missing parentheses: https://github.com/airbytehq/airbyte/pull/38799/commits/c9d2afb6d6d92667546a5a70232d6fe908c7628c

**Comment by bnchrch** [comment]
### Merge activity

* **Jun 12, 11:45 AM EDT**: @bnchrch started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/38799).
